app/models/user.rb

```diff
@@ -0,0 +1,16 @@
+class User < ApplicationRecord
+  # Include default devise modules. Others available are:
+  # :confirmable, :lockable, :timeoutable, :trackable and :omniauthable
+  devise :database_authenticatable, :registerable,
+         :recoverable, :rememberable, :validatable
+
+  has_many :tasks
+  has_many :lists, through: :tasks
+
+  delegate :completed_tasks, :incompleted_tasks, to: :tasks
+
+  def fetch_tasks(filter_tasks_by)
+    method = Task::FILTER_TASKS_BY_STATES.fetch(filter_tasks_by, :incompleted_tasks)
+    send(method)
```

Comment:
i'm not crazy about using the `send` method as it can obfuscate your code and make it harder to debug at times but i ended up deciding to use it here since i ensured that only 1 of three methods could be returned. 

Also, if the user manipulates the DOM and inputs invalid data into the `filter_tasks` param, line 13 will default the filter method to `incompleted_tasks`. 

I'm not completely sold this is the best approach but I thought code conciseness was an advantage here.